### ISSUE TITLE
Removing text annotations fails sometime

### ISSUE THREAD

**Issue by mcolacino** [insideout10/wordlift-plugin#234]
This slice of markup taken from http://www.lafabbricadellarealta.com/the-simplest-cheapest-and-most-often-overlooked-trick-that-every-consultant-should-know/ shows as **sometimes removing text annotations fails** saving into the database unuseful markup.
 
```
Their ideas are debated, evaluated and – more often than not – refused. In this article I <span id="urn:enhancement-127a941c-5bdc-1487-3511-f204b4e0d33c" class="textannotation"><span id="urn:enhancement-eca637f1-d6db-3d2f-b122-86272d2cc8ef" class="textannotation"><span id="urn:enhancement-961d102f-72db-fe00-6b67-f3abe1e3fbbb" class="textannotation"><span id="urn:enhancement-eac4af3a-4dd2-c92b-6f84-af835e3b7fa9" class="textannotation"><span id="urn:enhancement-fd3a256a-3fdf-0b8f-84c4-cc6e156a0d6c" class="textannotation">want<span></span></span></span></span></span></span> to provide you with my own recipe to take ego out of the equation and not only make rejection painless.
```

After a first check, **the issue seems to be related to the blank span just after the word "want"**. 
The issue can be easily reproduced.

**Comment by mcolacino** [comment]
Further tests show how text annotation removing fails when any additional markup is in place within a given text annotation. The bug happens also with single not-nested annotations.

**Comment by mcolacino** [comment]
Resolved as hotfix in 3.3.2. A better solution is that one referred by the issue #236
